File: photoeditor/src/test/java/ja/burhanrashid52/photoeditor/SaveSettingsTest.java
package ja.burhanrashid52.photoeditor;

import android.graphics.Bitmap;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.TestCase.assertTrue;


public class SaveSettingsTest {

    @Test
    public void testByDefaultTransparentAndClearViewFlagSettingIsEnabled() {
        SaveSettings saveSettings = new SaveSettings.Builder().build();
        assertTrue(saveSettings.isClearViewsEnabled());
        assertTrue(saveSettings.isTransparencyEnabled());
    }

    @Test
    public void testWhenTransparentSettingIsDisabled() {
        SaveSettings saveSettings = new SaveSettings.Builder()
                .setTransparencyEnabled(false)
                .build();

        assertFalse(saveSettings.isTransparencyEnabled());
        assertTrue(saveSettings.isClearViewsEnabled());
    }

    @Test
    public void testWhenClearViewAfterSaveSettingIsDisabled() {
        SaveSettings saveSettings = new SaveSettings.Builder()
                .setClearViewsEnabled(false)
                .build();

        assertFalse(saveSettings.isClearViewsEnabled());
        assertTrue(saveSettings.isTransparencyEnabled());
    }

    @Test
    public void testWhenBothTransparentClearViewAfterSaveSettingIsDisabled() {
        SaveSettings saveSettings = new SaveSettings.Builder()
                .setClearViewsEnabled(false)
                .setTransparencyEnabled(false)
                .build();

        assertFalse(saveSettings.isClearViewsEnabled());
        assertFalse(saveSettings.isTransparencyEnabled());
    }

    @Test
    public void testDefaultCompressAndQualitySaveSettings() {
        SaveSettings saveSettings = new SaveSettings.Builder()
                .build();

        assertEquals(saveSettings.getCompressFormat(), Bitmap.CompressFormat.PNG);
        assertEquals(saveSettings.getCompressQuality(), 100);
    }

    @Test
    public void testCompressAndQualitySaveSettingsValues() {

        Bitmap.CompressFormat compressFormat = Bitmap.CompressFormat.PNG;
        int compressQuality = 50;

        SaveSettings saveSettings = new SaveSettings.Builder()
                .setCompressFormat(compressFormat)
                .setCompressQuality(compressQuality)
                .build();

        assertEquals(saveSettings.getCompressFormat(), compressFormat);
        assertEquals(saveSettings.getCompressQuality(), compressQuality);
    }
}

File: photoeditor/src/test/java/ja/burhanrashid52/photoeditor/SaveSettingsTest.kt
package ja.burhanrashid52.photoeditor

import junit.framework.TestCase
import android.graphics.Bitmap.CompressFormat
import junit.framework.TestCase.assertEquals
import junit.framework.TestCase.assertFalse
import org.junit.Test

class SaveSettingsTest {
    @Test
    fun testByDefaultTransparentAndClearViewFlagSettingIsEnabled() {
        val saveSettings = SaveSettings.Builder().build()
        TestCase.assertTrue(saveSettings.isClearViewsEnabled)
        TestCase.assertTrue(saveSettings.isTransparencyEnabled)
    }

    @Test
    fun testWhenTransparentSettingIsDisabled() {
        val saveSettings = SaveSettings.Builder()
            .setTransparencyEnabled(false)
            .build()
        assertFalse(saveSettings.isTransparencyEnabled)
        TestCase.assertTrue(saveSettings.isClearViewsEnabled)
    }

    @Test
    fun testWhenClearViewAfterSaveSettingIsDisabled() {
        val saveSettings = SaveSettings.Builder()
            .setClearViewsEnabled(false)
            .build()
        assertFalse(saveSettings.isClearViewsEnabled)
        TestCase.assertTrue(saveSettings.isTransparencyEnabled)
    }

    @Test
    fun testWhenBothTransparentClearViewAfterSaveSettingIsDisabled() {
        val saveSettings = SaveSettings.Builder()
            .setClearViewsEnabled(false)
            .setTransparencyEnabled(false)
            .build()
        assertFalse(saveSettings.isClearViewsEnabled)
        assertFalse(saveSettings.isTransparencyEnabled)
    }

    @Test
    fun testDefaultCompressAndQualitySaveSettings() {
        val saveSettings = SaveSettings.Builder()
            .build()
        assertEquals(saveSettings.compressFormat, CompressFormat.PNG)
        assertEquals(saveSettings.compressQuality, 100)
    }

    @Test
    fun testCompressAndQualitySaveSettingsValues() {
        val compressFormat = CompressFormat.PNG
        val compressQuality = 50
        val saveSettings = SaveSettings.Builder()
            .setCompressFormat(compressFormat)
            .setCompressQuality(compressQuality)
            .build()
        assertEquals(saveSettings.compressFormat, compressFormat)
        assertEquals(saveSettings.compressQuality, compressQuality)
    }
}